src/shell/lisa_shell

```diff
@@ -93,10 +93,49 @@ fi
 echo -ne "$LISASHELL_DEFAULT"
 }
 
+function _lisa-update-all {
+echo -ne "${LISASHELL_BLUE}"
+echo -e "*** Update LISA installation"
+
+git update-index -q --refresh
+ret=$?
+if [ $ret -ne 0 ]; then
+    echo "lisa internal error: git update-index failed.  Please report it: https://github.com/ARM-software/lisa/issues"
+    return $ret
+fi
+
+if ! git diff-index --quiet HEAD; then
+    echo "There are outstanding uncommitted changes."
+    echo "Please, commit your changes or stash them before you can update lisa"
+    return 1
+fi
+
+curr_commit=$(git rev-parse HEAD)
+if ! git merge-base --is-ancestor $curr_commit origin/master; then
+    echo "You have committed changes that are not part of origin/master"
+    echo "Please move to the master branch"
+    return 1
+fi
+
+git pull --ff-only origin master
```

Comment:
You're right, I don't like the hardcoding of origin.  I guess it could be changed to:
``` bash
origin_name=$(git remote -v | grep ARM-software/lisa.git | grep -m 1 fetch | awk '{print $1}')
```

Do you prefer that?